src/Uno.UI/UI/Xaml/Media/Animation/Animators/GPUFloatValueAnimator.iOSmacOS.cs

```diff
@@ -417,22 +440,80 @@ private NSValue ToCASkewTransform(float angleX, float angleY)
 			return NSValue.FromCATransform3D(CATransform3D.MakeFromAffine(matrix));
 		}
 
+		private static void OnCoreWindowVisibilityChanged(CoreWindow sender, VisibilityChangedEventArgs args)
+		{
+			lock(_weakActiveInstanceCache)
+			{
+				foreach (var p in _weakActiveInstanceCache.ToArray())
```

Comment:
As you're locking, I would think there's no need to convert to an array.

Reply:
this the weird part, despite the lock on etor+add/remove, i still get the collection got modified during iteration exception...

Reply:
This would mean that animations get changed by raising the event then :) That's fine, let's keep it that way for now.